src/ansys/additive/core/additive.py

```diff
@@ -402,7 +402,7 @@ def _simulate(
                     if progress.state == ProgressState.ERROR:
                         raise Exception(progress.message)
                 if response.HasField("melt_pool"):
-                    return SingleBeadSummary(input, response.melt_pool)
+                    return self._check_for_thermal_history(response, input)
```

Comment:
There's a design smell here. `check_for_*` should either return a bool or it should return void upon success and throw an exception if the check fails. There are better ways to do what you're trying to do. Here's one approach. First update the protobuf definitions to add `optional` to the field.
```
message MeltPool {
    repeated MeltPoolTimeStep time_steps = 1;
    optional string thermal_history_vtk_zip = 2;
}
```
Then, at this point in the code, add `if response.meltpool.has_thermal_history_vtk_zip:` followed by the download of the zip file. After this, call `return SingleBeadSummary(input, response.melt_pool, path_to_zip_file)` and pass in the zip file, NOT the folder the contents were extracted to. Do the extraction in the `init()` method of `SingleBeadSummary`.

Reply:
Okay, I see that the protobuf definitions are optional by default. So will try to organize the code in a way that the `additive.py` just downloads the zip file, but the `SingleBeadSummary` takes care of extracting the contents.